test/zipkin/span_context_test.cc

```diff
@@ -0,0 +1,152 @@
+#include "zipkin/span_context.h"
+#include "zipkin/zipkin_core_constants.h"
+
+#include "gtest/gtest.h"
+
+namespace Zipkin {
+
+TEST(ZipkinSpanContextTest, populateFromString) {
+  SpanContext spanContext;
+
+  // Non-initialized span context
+  EXPECT_EQ(0ULL, spanContext.trace_id());
+  EXPECT_EQ("0000000000000000", spanContext.traceIdAsHexString());
+  EXPECT_EQ(0ULL, spanContext.id());
+  EXPECT_EQ("0000000000000000", spanContext.idAsHexString());
+  EXPECT_EQ(0ULL, spanContext.parent_id());
+  EXPECT_EQ("0000000000000000", spanContext.parentIdAsHexString());
+  EXPECT_FALSE(spanContext.isSetAnnotation().cr);
+  EXPECT_FALSE(spanContext.isSetAnnotation().cs);
+  EXPECT_FALSE(spanContext.isSetAnnotation().sr);
+  EXPECT_FALSE(spanContext.isSetAnnotation().ss);
+  EXPECT_EQ("0000000000000000;0000000000000000;0000000000000000", spanContext.serializeToString());
+
+  // Span context populated with trace id, id, parent id, and no annotations
```

Comment:
I have added a test to verify this  behavior. Note that the Span class has a member named traceIdHigh_, plus a flag indicating whether or not it has been set. That's how a Span with a 128-bit trace id can be created. Currently, the code ignores the higher bits.
Here is the  test I added:
https://github.com/amalgam8/envoy/blob/zipkin/test/zipkin/span_context_test.cc#L117